lib/hasPermissions.ts

```diff
@@ -203,16 +207,44 @@ export class HasPermissionsDirectiveVisitor<
     const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
       value: unknown,
       _: unknown,
-      __: unknown,
+      container: GraphQLArgument | GraphQLInputObjectType | GraphQLObjectType,
       context: TContext,
       resolverInfo: Record<string, unknown>,
       resolverSource: unknown,
       resolverArgs: Record<string, unknown>,
+      path?: Array<string>,
     ): unknown => {
+      if (
+        location ===
+          (DirectiveLocation.INPUT_FIELD_DEFINITION ||
+            DirectiveLocation.INPUT_OBJECT) &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        value === container.getFields()[path[path.length - 1]].defaultValue
+      ) {
+        return value;
+      }
+
+      if (
+        location === DirectiveLocation.ARGUMENT_DEFINITION &&
```

Comment:
The conditions in here are the same from the if above. Just move this `location === DirectiveLocation.ARGUMENT_DEFINITION` to the `||` in the if above